File: API/PageAndRDL/Export/C#/Export/ExportForm.cs
using GrapeCity.ActiveReports;
using GrapeCity.ActiveReports.Extensibility.Rendering;
using GrapeCity.ActiveReports.Rendering.IO;
using System;
using System.Collections.Specialized;
using System.IO;
using System.Text;
using System.Windows.Forms;

namespace Export
{
	public partial class ExportForm : Form
	{
		private const string _json = "json";
		private const string _pdf = "pdf";
		private const string _xlsx = "xlsx";
		private const string _docx = "docx";
		private const string _csv = "csv";
		private const string _mht = "mht";

		public ExportForm()
		{
			InitializeComponent();
			FillReportsNames();
		}

		private void ExportForm_Load(object sender, EventArgs e)
		{
			reportsNames.SelectedIndex = 0;
			exportTypes.SelectedIndex = 0;
		}

		private void FillReportsNames()
		{
			var directoryInfo = new DirectoryInfo("../../../../Reports/");
			foreach (var file in directoryInfo.GetFiles())
			{
				reportsNames.Items.Add(file.Name);
			}
		}

		private void exportButton_Click(object sender, EventArgs e)
		{
			var reportName = (string)reportsNames.SelectedItem;
			var exportType = (string)exportTypes.SelectedItem;
			IRenderingExtension renderingExtension = null;
			NameValueCollection settings = null;
			var exporTypeLower = exportType.ToLower();
			switch (exporTypeLower)
			{
				case _pdf:
					renderingExtension = new GrapeCity.ActiveReports.Export.Pdf.Page.PdfRenderingExtension();
					settings = new GrapeCity.ActiveReports.Export.Pdf.Page.Settings();
					break;
				case _xlsx:
					renderingExtension = new GrapeCity.ActiveReports.Export.Excel.Page.ExcelRenderingExtension();
					ISettings excelSettings = new GrapeCity.ActiveReports.Export.Excel.Page.ExcelRenderingExtensionSettings()
					{
						FileFormat = GrapeCity.ActiveReports.Export.Excel.Page.FileFormat.Xlsx
					};
					settings = excelSettings.GetSettings();
					break;
				case _csv:
					settings = new GrapeCity.ActiveReports.Export.Text.Page.CsvRenderingExtension.Settings()
					{
						ColumnsDelimiter = ",",
						RowsDelimiter = "\r\n",
						QuotationSymbol = '"',
						Encoding = Encoding.UTF8
					};
					renderingExtension = new GrapeCity.ActiveReports.Export.Text.Page.CsvRenderingExtension();
					break;
				case _docx:
					renderingExtension = new GrapeCity.ActiveReports.Export.Word.Page.WordRenderingExtension();
					settings = new GrapeCity.ActiveReports.Export.Word.Page.Settings() { FileFormat = GrapeCity.ActiveReports.Export.Word.Page.FileFormat.OOXML };

					break;
				case _mht:
					renderingExtension = new GrapeCity.ActiveReports.Export.Html.Page.HtmlRenderingExtension();
					settings = new GrapeCity.ActiveReports.Export.Html.Page.Settings(){ MhtOutput = true, OutputTOC = true, Fragment = false };

					break;
				case _json:
					settings = new GrapeCity.ActiveReports.Export.Text.Page.JsonRenderingExtension.Settings() { Formatted = true };
					renderingExtension = new GrapeCity.ActiveReports.Export.Text.Page.JsonRenderingExtension();
					break;
			}

			var report = new PageReport(new FileInfo(@"..\..\..\..\Reports\" + reportName));
			var fileName =  Path.GetFileNameWithoutExtension(reportName);
			var saveFileDialog = new SaveFileDialog()
			{
				FileName = fileName + GetSaveDialogExtension(exporTypeLower),
				Filter = GetSaveDialogFilter(exporTypeLower)
			};

			if (saveFileDialog.ShowDialog() != DialogResult.OK)
			{
				return;
			}

			var outputDirectory = new DirectoryInfo(Path.GetDirectoryName(saveFileDialog.FileName));
			var outputProvider = new FileStreamProvider(outputDirectory, Path.GetFileNameWithoutExtension(saveFileDialog.FileName));

			outputProvider.OverwriteOutputFile = true;

			exportButton.Enabled = false;
			report.Document.Render(renderingExtension, outputProvider, settings);
			exportButton.Enabled = true;
		}

		private string GetSaveDialogFilter(string exportType)
		{
			return exportType + " files | *." + exportType;
		}

		private string GetSaveDialogExtension(string exportType)
		{
			return "." + exportType;
		}
	}
}

File: API/PageAndRDL/Export/VB.NET/Export/ExportForm.vb
Imports System.Collections.Specialized
Imports System.IO
Imports System.Text
Imports GrapeCity.ActiveReports
Imports GrapeCity.ActiveReports.Extensibility.Rendering

Public Class ExportForm

	Private Sub FillReportsNames()
		Dim directoryInfo As Object = New DirectoryInfo("../../../../Reports/")
		For Each file As Object In directoryInfo.GetFiles()
			reportsNames.Items.Add(file.Name)
		Next
	End Sub

	Private Sub exportButton_Click(sender As Object, e As EventArgs) Handles exportButton.Click

		Const _json As String = "json"
		Const _pdf As String = "pdf"
		Const _xlsx As String = "xlsx"
		Const _docx As String = "docx"
		Const _csv As String = "csv"
		Const _mht As String = "mht"

		Dim reportName = CType(reportsNames.SelectedItem, String)
		Dim exportType = CType(exportTypes.SelectedItem, String)
		Dim renderingExtension As IRenderingExtension = Nothing
		Dim settings As NameValueCollection = Nothing

		Dim exporTypeLower As String = exportType.ToLower()
		Select Case exporTypeLower
			Case _pdf
				renderingExtension = New GrapeCity.ActiveReports.Export.Pdf.Page.PdfRenderingExtension()
				settings = New GrapeCity.ActiveReports.Export.Pdf.Page.Settings()
				Exit Select
			Case _xlsx
				renderingExtension = New GrapeCity.ActiveReports.Export.Excel.Page.ExcelRenderingExtension()
				Dim excelSettings As ISettings = New GrapeCity.ActiveReports.Export.Excel.Page.ExcelRenderingExtensionSettings() With {
					.FileFormat = GrapeCity.ActiveReports.Export.Excel.Page.FileFormat.Xlsx
				}
				settings = excelSettings.GetSettings()
				Exit Select
			Case _csv
				settings = New GrapeCity.ActiveReports.Export.Text.Page.CsvRenderingExtension.Settings() With {
					.ColumnsDelimiter = ",",
					.RowsDelimiter = vbCr & vbLf,
					.QuotationSymbol = """"c,
					.Encoding = Encoding.UTF8
				}

				renderingExtension = New GrapeCity.ActiveReports.Export.Text.Page.CsvRenderingExtension()
				Exit Select
			Case _docx
				renderingExtension = New GrapeCity.ActiveReports.Export.Word.Page.WordRenderingExtension()
				settings = New GrapeCity.ActiveReports.Export.Word.Page.Settings() With {
					.FileFormat = GrapeCity.ActiveReports.Export.Word.Page.FileFormat.OOXML
				}
				Exit Select
			Case _mht
				renderingExtension = New GrapeCity.ActiveReports.Export.Html.Page.HtmlRenderingExtension()
				settings = New GrapeCity.ActiveReports.Export.Html.Page.Settings() With {
					.MhtOutput = True,
					.OutputTOC = True,
					.Fragment = False
				}
				Exit Select
			Case _json
				settings = New GrapeCity.ActiveReports.Export.Text.Page.JsonRenderingExtension.Settings() With {
					.Formatted = True
				}
				renderingExtension = New GrapeCity.ActiveReports.Export.Text.Page.JsonRenderingExtension()
				Exit Select
		End Select

		Dim report = New PageReport(New FileInfo("..\..\..\..\Reports\" + reportName))
		Dim fileName = Path.GetFileNameWithoutExtension(reportName)
		Dim saveFileDialog = New SaveFileDialog()
		saveFileDialog.FileName = fileName + GetSaveDialogExtension(exporTypeLower)
		saveFileDialog.Filter = GetSaveDialogFilter(exporTypeLower)

		If saveFileDialog.ShowDialog() <> DialogResult.OK Then
			Return
		End If

		Dim outputDirectory As Object = New DirectoryInfo(Path.GetDirectoryName(saveFileDialog.FileName))
		Dim outputProvider As Object = New Rendering.IO.FileStreamProvider(outputDirectory, Path.GetFileNameWithoutExtension(saveFileDialog.FileName))

		outputProvider.OverwriteOutputFile = True

		exportButton.Enabled = False
		report.Document.Render(renderingExtension, outputProvider, settings)
		exportButton.Enabled = True
	End Sub

	Private Sub ExportForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
		FillReportsNames()
		reportsNames.SelectedIndex = 0
		exportTypes.SelectedIndex = 0
	End Sub

	Private Function GetSaveDialogFilter(exportType As String) As String
		Return exportType + " files | *." + exportType
	End Function

	Private Function GetSaveDialogExtension(exportType As String) As String
		Return "." + exportType
	End Function

End Class
